packages/router/src/router-node.ts

```diff
@@ -147,8 +147,12 @@ export interface PageProps<P> {
   children?: JSX.Element;
 }
 
+type TPageProps<U> = U extends LoadResult<infer P> ? PageProps<P> : PageProps<never>;
+
 export type InferPageProps<T extends (...args: any[]) => any> =
- ReturnType<T> extends LoadResult<infer P> ? PageProps<P> : PageProps<never>;
+ T extends Load<infer R> ?
+  R extends Promise<infer A> ? PageProps<A> : PageProps<R> :
```

Comment:
`Awaited<T>` is probably better? so you don't have to unwrap the `Promise`. 

If it's unideal, maybe this works too:

```ts
T extends Load<infer R> ? PageProps<R extends Promise<infer A> ? A : R>
```